[basic] Replace 'could' and 'might' (#4365)
cplusplus/draft: source/basic.tex

a specialization of a template
whose (possibly instantiated) declaration is an exposure.
\begin{note}

> A specialization can be produced by implicitly or explicitly instantiation.

Reviewer: Should these adverbs become adjectives?

Author: fixed

lifetime ends there are significant restrictions on the use of the
object, as described below, in~\ref{class.base.init} and
in~\ref{class.cdtor}. Also, the behavior of an object under construction

> and destruction is possibly not the same as the behavior of an object whose

Reviewer: This is very nice. We should give that as an example of possible rewordings.

When \tcode{D} is the type of a complete object, it will have a subobject of
type \tcode{B}, so it must be aligned appropriately for a \tcode{long double}.
If \tcode{D} appears as a subobject of another object that also has \tcode{B}
as a virtual base class, it is possible for the \tcode{B} subobject to be part of a different

> subobject, reducing the alignment requirements on the \tcode{D} subobject.

Author: fixed

\begin{note}
A pointer past the end of an object\iref{expr.add}
is not considered to point to an unrelated object
of the object's type,

> even if the unrelated object would be located at that address.

Reviewer: Can we just say "is located" here? What does the extra qualification achieve here?

Author: fixed

\begin{example}
A library I/O function that blocks until the I/O operation is complete can
be considered to continuously check whether the operation is complete. Each

> such check possibly consists of one or more execution steps, for example using

Reviewer 2: What is the alternative? (If it doesn't consist of one or more execution steps, then what does it do?) Can we drop the "possibly"?

Author: dropped

guarantees, which in turn are stronger than weakly parallel forward progress
guarantees.
\begin{note}
For example, some kinds of synchronization between threads of execution would only

> make progress if the respective threads of execution provide parallel forward progress

Reviewer: I feel like this should now say "if the respective threads ... _provided_ parallel forward progress guarantees." What do you think?

Author: of course
fixed

guarantees, which in turn are stronger than weakly parallel forward progress
guarantees.
\begin{note}

> For example, some kinds of synchronization between threads of execution would only
> make progress if the respective threads of execution provided parallel forward progress
> guarantees, but will fail to make progress under weakly parallel guarantees.

Reviewer 2: Is this really right? Should this instead say:
```suggestion
For example, some kinds of synchronization between threads of execution are only guaranteed to
make progress if the respective threads of execution provide parallel forward progress
guarantees, but will not necessarily make progress under weakly parallel guarantees.
```
